content/getting-started/solving-kata.md

```diff
@@ -6,10 +6,66 @@ next: "/getting-started/solutions/"
+
+When tests output panel turns green and your solution passes all the tests, congratulations! Task is solved, and you are awarded with Honor points and your progress increases (see [following paragraph](#rewards) for more details).
+
+### Submit Final: Production Quality Code
+
+Some users like to write code golfed or creative/clever solutions, but most try to write production quality code. After you have completed all of the test cases (via the attempt button), you will be given an opportunity to cleanup your code so that it's "code review" ready. After clicking **"Submit Final"** your solution is stored in the system and available for others to see, comment, and vote.
+
+## When tests fail over again
+
+Sometimes you just cannot get your solution right. Output panel is red after every attempt, kata might appear broken, or you receive some errors you do not understand and you have no idea what's going on. But don't worry, there are many ways to get help. See [Troubleshooting your Solution (TODO: add link to FAQ)]() FAQ to get some advice.
+
+## Unlocking solutions
+
+Some kata are difficult. Some of them turn out to be more difficult than they initially appeared. Sometimes it might just not be your good day. It's OK, and happens to all of us. You can take a break from the kata and try another one, you will always be able to come back to this one later. You can do more research, search the web for algorithms, or ask friends for help. But if nothing helps, and you _really_ need to know what other users came up with to defeat the task, you can unlock solutions without finishing the kata.
+
+When you unlock solutions you immediately gain access to all information about kata just as if you have finished it: solutions of other users, code of full test suite, etc. Although be warned, because this way you get no reward, and you forfeit all eligibility for any reward you would get for solving this kata in the future, for example in another language. Use this option with care, giving up is also a way to learn on Codewars, but sometimes waiting a bit and doing more research also helps.
+
+## Rewards
+
+You worked hard to solve your last challenge, and after you succeeded, you definitely deserve a reward. Each solved tasks rewards you in two ways: progressing your rank, and granting some amount of Honor points. The more difficult kata you solve, the bigger the reward.
+
+### Rank
+
+Rank level reflects your experience and knowledge you have gained by solving kata. You start at the lowest rank of 8 kyū, and each correct solution brings you closer to a higher rank. You can see your rank progress toward next level on your profile page:
+
+![rank progress](./img/solving_01_rank-progress.png)
+
+More information on ranks and progress can be found [here (TODO: add link to docs on progress)]().
+
+### Honor
```

Comment:
you so much don't like honor that you didn't even show the top right corner of the page to show where the info is visible to the user while you showed the progress wheel... XD

;)

=> add a picture?

Reply:
I did not want to put too many pictures here, because I think it's easily visible even for newbies, and that actual details will be present in "UI docs", possibly linked from here